src/profilers.cpp

```diff
@@ -61,11 +61,23 @@ namespace aperf {
                    fs::path result_processed,
                    bool capture_immediately) {
 
-    this->result_out = result_out;
     std::string instrs = connection_instrs.get_instructions(this->get_thread_count());
 
+    //sanitize metric name for log file name
+    std::regex invalidChars(R"([\/:*?"<>|])");
+
+    std::string sanitized_metric_name = std::regex_replace(this->metric_name, invalidChars, "_");
+
+    sanitized_metric_name = std::regex_replace(sanitized_metric_name, std::regex(R"(^\s+|\s+$)"), "");
+
+    if (sanitized_metric_name.empty()) {
+        sanitized_metric_name = "metric_command_stderr.log";
+    }else{
+      sanitized_metric_name += "_stderr.log";
+    }
```

Comment:
I would also sanitise whitespace characters (it's not strictly necessary, but handling filenames without spaces is easier in general) and replace longer sequences of invalid characters with just one underscore.